pkg/apis/core/validation/validation.go

```diff
@@ -3899,45 +3904,90 @@ func validateSysctls(sysctls []core.Sysctl, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
-// ValidatePodSecurityContext test that the specified PodSecurityContext has valid data.
-func ValidatePodSecurityContext(securityContext *core.PodSecurityContext, spec *core.PodSpec, specPath, fldPath *field.Path) field.ErrorList {
```

Comment:
the delta here is *really* hard to review... to be confident this isn't changing any existing validation, I would recommend:

* keep ValidatePodSecurityContext as-is
* separately forbid values in OS-specific fields if spec.OS is set to an OS which does not permit them

Reply:
essentially, the OS-specific validation should be *additive* to the existing validation, should only run when spec.OS is specified, and should focus solely on forbidding values in OS-specific fields that don't match the specified pod OS